t/ui/26-jobs_restart.t

```diff
@@ -73,23 +73,86 @@ sub prepare_database {
+        $driver->find_element('#restart-button-options')->click;
+        my $restart_parent = $driver->find_element('#restart-result-skip-parents');
+        ok $restart_parent->is_displayed, 'advanced restart entries shown for child';
+        $restart_parent->attribute_like('href', qr/skip_parents=1/, 'skip parents API URL correct for child');
+        $driver->find_element('#restart-result-skip-ok-children')
+          ->attribute_like('href', qr/skip_ok_result_children=1/, 'skip OK children API URL correct for child');
+    };
+    subtest 'assets missing; there is no parent' => sub {
+        is($driver->get('/tests/99939'), 1, 'go to job 99939');
+        is @{$driver->find_elements('//a[contains(@id, "restart")]', 'xpath')}, 1, 'only one restart action found';
+        $driver->find_element('#restart-result')->click();
+        wait_for_ajax(msg => 'fail to start job with missing asset and no parent');
+        like($driver->get_current_url, qr|tests/99939|, 'no auto refresh when there are errors/warnings');
+        like(
+            $driver->find_element('#flash-messages')->get_text,
+            qr/Job 99939 misses.*\.iso.*Ensure to provide mandatory assets/s,
+            'restarting job with missing asset results in an error'
+        );
+    };
+    subtest 'assets missing; there is a parent' => sub {
+        $schema->resultset('JobDependencies')
+          ->create(
+            {child_job_id => 99939, parent_job_id => 99947, dependency => OpenQA::JobDependencies::Constants::CHAINED});
+        is($driver->get('/tests/99939'), 1, 'go to job 99939');
+        $driver->find_element('#restart-result')->click();
+        wait_for_ajax(msg => 'fail to start job with missing asset and no parent');
+        like(
+            $driver->find_element('#flash-messages')->get_text,
+            qr/Job 99939 misses.*\.iso.*You may try to retrigger the parent job/s,
+            'restarting job with missing asset results in an error'
+        );
+    };
+    subtest 'force restart' => sub {
+        update_last_job_id;
+        $driver->find_element('#flash-messages button.force-restart')->click();
+        wait_for_ajax(msg => 'forced job restart');
+        like($driver->find_element_by_link_text('new job')->get_attribute('href'),
+            expected_job_id_regex, 'warning with link to new job appears');
+    };
+    subtest 'successful restart' => sub {
+        is($driver->get('/tests/99946'), 1, 'go to job 99946');
+        update_last_job_id;
+        $driver->find_element('button > #restart-result')->click();
+        wait_for_ajax(msg => 'successful restart from info panel in test results');
+        like($driver->get_current_url(), expected_job_id_regex, 'auto refresh to restarted job');
+        like($driver->find_element('#info_box .card-header')->get_text(),
+            qr/textmode\@32bit/, 'restarted job is correct');
```

Comment:
re https://github.com/os-autoinst/openQA/pull/4417#issuecomment-999355001

I wonder where the gap is - the test looks to cover it all? What's missing?

Reply:
I guess what the test is missing that after clicking a restart button we should be automatically moved to the new clone, same as for the default button action